### DISCUSSION TITLE
[bug?] update_all lost data when using with $addToSet

### DISCUSSION THREAD

**Discussion by midnight-wonderer** [mongodb/mongoid#5674] [General]
`$addToSet` is weird.

Here is the model:
~~~ruby
class BuggyModel
  include ::Mongoid::Document
  store_in collection: 'buggy_models'

  field :list, type: :array, default: []
end
~~~

Create one element:
~~~ruby
::BuggyModel.create!({ _id: 'test' })
~~~

Add something to the set:
~~~ruby
::BuggyModel.update_all({
  '$addToSet': {
    list: 'a',
  },
})
~~~

Reload:
~~~ruby
::BuggyModel.find('test').list
~~~

Expecting: `['a']`  
Actual: `[nil]`

Why?

**Comment by midnight-wonderer** [comment]
Digging further, the problem is in `mongoize` of `Array` fields that was called from `hash.__consolidate__`.

The workaround exists by omitting the type of the field.
Changing this:
~~~ruby
field :list, type: :array, default: []
~~~
To this:
~~~ruby
field :list, default: []
~~~

Probably `Mongoid::Persistable::LIST_OPERATIONS` handling needs revises.

**Comment by jamis** [comment]
Thank you for reporting this! I don't know how your post got past our radar, but I apologize for missing this. I've added a ticket to our tracking system here, if you'd like to follow along: https://jira.mongodb.org/browse/MONGOID-5686

**Comment by comandeo** [comment]
Since the issue was fixed, I am closing this discussion. @midnight-wonderer thank you again for reporting this!
